src/DataTable/README.md

```diff
@@ -1569,4 +1569,201 @@ You can create your own cell content by passing the `Cell` property to a specifi
     </DataTable>
   );
 }
+
+```
+##  maxSelectedRows and onMaxSelectedRows props
+
+These props will allow us to handle the maximum number of selectable rows, which is necessary for validation. Using the `maxSelectedRows` prop, the implementation process can be simplified. You only need to pass the maximum number of rows you want to have selected.
+
+After selecting the maximum possible number of rows, you can display an error message or perform other actions. This is achieved through the  `onMaxSelectedRows`  callback. In the example below, the callback will be executed when the table has 3 rows selected.
+
+```jsx live
+() => {
+  return (
+    <DataTable
+      isPaginated
+      isSelectable
+      maxSelectedRows={3}
+      onMaxSelectedRows={() => console.log('this is the last row allowed')}
+      itemCount={7}
+      data={[
+        {
+          name: 'Lil Bub',
+          color: 'brown tabby',
+          famous_for: 'weird tongue',
+        },
+        {
+          name: 'Grumpy Cat',
+          color: 'siamese',
+          famous_for: 'serving moods',
+        },
+        {
+          name: 'Smoothie',
+          color: 'orange tabby',
+          famous_for: 'modeling',
+        },
+        {
+          name: 'Maru',
+          color: 'brown tabby',
+          famous_for: 'being a lovable oaf',
+        },
+        {
+          name: 'Keyboard Cat',
+          color: 'orange tabby',
+          famous_for: 'piano virtuoso',
+        },
+        {
+          name: 'Long Cat',
+          color: 'russian white',
+          famous_for: 'being looooooooooooooooooooooooooong',
+        },
+        {
+          name: 'Zeno',
+          color: 'brown tabby',
+          famous_for: 'getting halfway there',
+        },
+      ]}
+      columns={[
+        {
+          Header: 'Name',
+          accessor: 'name',
+        },
+        {
+          Header: 'Famous For',
+          accessor: 'famous_for',
+        },
+        {
+          Header: 'Coat Color',
+          accessor: 'color',
+          filter: 'includesValue',
+          filterChoices: [
+            {
+              name: 'russian white',
+              number: 1,
+              value: 'russian white',
+            },
+            {
+              name: 'orange tabby',
+              number: 2,
+              value: 'orange tabby',
+            },
+            {
+              name: 'brown tabby',
+              number: 3,
+              value: 'brown tabby',
+            },
+            {
+              name: 'siamese',
+              number: 1,
+              value: 'siamese',
+            },
+          ],
+        },
+      ]}
+    >
+      <DataTable.TableControlBar />
+      <DataTable.Table />
+      <DataTable.EmptyTable content="No results found" />
+      <DataTable.TableFooter />
+    </DataTable>
+  );
+}
 ```
+#### example with react-intl
```

Comment:
@johnvente What is the intent of this example with `react-intl`? I'm not quite sure what value it provides to consumers. What is this example trying to demonstrate for consumers? My two cents is that this particular example is unnecessary.

Also, as is, it results in a JS error within the example code block preview:

<img width="938" alt="image" src="https://github.com/openedx/paragon/assets/2828721/aca0d2bd-25eb-4196-a7c3-112f772cabf0">

Reply:
@adamstankiewicz Probably this is unnecessary I'll remove it